### PR TITLE
Improve documentation for TransformStamped.msg

### LINKED ISSUE #94: Add documentation to TransformStamped as to why it has the extra field
This is a follwup to the [pre Foxy Message API review](https://github.com/ros2/common_interfaces/pull/86) 

https://github.com/ros2/common_interfaces/pull/86/files#r392491840

### PR DESCRIPTION
**Pull request by tfoote**
Fixes #94

### CODE REVIEW

File: geometry_msgs/msg/TransformStamped.msg [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-# to the coordinate frame child_frame_id
+# to the coordinate frame child_frame_id at the time of header.stamp
 #
 # This message is mostly used by the
-# <a href="http://www.ros.org/wiki/tf">tf</a> package.
+# <a href="http://www.ros.org/wiki/tf2">tf2</a> package.
 # See its documentation for more information.
+#
+# The child_frame_id is necessary in addition to the frame_id
+# in the Header to communication the full reference for the transform
```
Comment (hidmic) [inline geometry_msgs/msg/TransformStamped.msg:9]: @tfoote 
```suggestion
# in the Header to communicate the full reference for the transform
```

Comment (hidmic) [reply geometry_msgs/msg/TransformStamped.msg:9]: I do wonder if this paragraph isn't just redundant with the definition of transform and the first paragraph.

File: geometry_msgs/msg/TransformStamped.msg [hunk 2]
```diff
@@ -1,9 +1,13 @@
 # This expresses a transform from coordinate frame header.frame_id
-# to the coordinate frame child_frame_id
+# to the coordinate frame child_frame_id at the time of header.stamp
 #
 # This message is mostly used by the
-# <a href="http://www.ros.org/wiki/tf">tf</a> package.
+# <a href="http://www.ros.org/wiki/tf2">tf2</a> package.
```
Comment (hidmic) [inline geometry_msgs/msg/TransformStamped.msg:5]: @tfoote hmm are we reusing the ROS wiki for ROS 2?

Comment (tfoote) [reply geometry_msgs/msg/TransformStamped.msg]: Good point updated to use index.ros.org

File: geometry_msgs/msg/TransformStamped.msg (hunk 1 above)
Comment (tfoote) [reply geometry_msgs/msg/TransformStamped.msg]: It is somewhat redundant but people asked for clarification in the API review so I figured another sentence could help.

**Pull request merged**